cfme/tests/cloud_infra_common/test_policy_simulation.py

```diff
@@ -0,0 +1,62 @@
+# -*- coding: utf-8 -*-
+"""This module tests events that are invoked by Cloud/Infra VMs."""
+import pytest
+
+from cfme.cloud.provider import CloudProvider
+from cfme.common.vm_views import PolicySimulationDetailsView
+from cfme.common.vm_views import PolicySimulationView
+from cfme.infrastructure.provider import InfraProvider
+from cfme.markers.env_markers.provider import ONE_PER_CATEGORY
+from cfme.utils.appliance.implementations.ui import navigate_to
+from cfme.utils.blockers import BZ
+
+pytestmark = [
+    pytest.mark.usefixtures('uses_infra_providers', 'uses_cloud_providers', 'setup_provider'),
+    pytest.mark.tier(2),
+    pytest.mark.provider(
+        classes=[InfraProvider, CloudProvider],
+        required_fields=[(['cap_and_util', 'chargeback_vm'], 'cu-24x7')],
+        selector=ONE_PER_CATEGORY,
+        scope="module"
+    ),
+]
+
+
+@pytest.mark.parametrize("navigation", ["provider_vms", "vm_summary"])
+def test_policy_simulation_ui(appliance, provider, navigation):
+    """
+    Bugzillas:
+        * 1670456, 1686617, 1686619, 1688359, 1550503
+    Polarion:
+        assignee: jdupuy
+        casecomponent: Control
+        caseimportance: medium
+        initialEstimate: 1/6hr
+    """
+    collection = provider.appliance.provider_based_collection(provider)
+    vm = collection.instantiate('cu-24x7', provider)
```

Comment:
Required: 
@john-dupuy, I was also doing this in same way but now I using `all` method.

1. Need to avoid this hard code.
```
vm = collection.all()[0]
```
2. If `vm` not available then we need to create it or `skip` test.
3. It's `setup` part for this test. May be in future you would like to add more test related to `policy` so better to put this as `fixture`.

Reply:
@digitronik I have a couple of thoughts regarding this. 
1) In this test, I am specifically using the providers that have this VM on them. 
2) It is slower to first get a whole list of all the VMs through navigation and then select the first item from that list. Leaving the hard code is faster. 
3. I agree about putting this into a fixture.

Reply:
@john-dupuy agreed... please for now put `all` method. It am also facing same problem it took time due to UI iteration... I moved most of `all` method from `UI` to `REST`. I am going to move `vm` `all` method to `rest`. It is in my task. Please wait... for some time.